### PR TITLE
Travis-CI compilation fixes

### PR DESCRIPTION
**Pull request by peabody124**
Fix some small bugs in the code that trigger warnings caught by newer versions of
arm-gcc

### CODE REVIEW

File: flight/Modules/PicoC/inc/expression.c [hunk 1]
```diff
@@ -420,7 +420,7 @@ void ExpressionAssign(struct ParseState *Parser, struct Value *DestValue, struct
             break;
         
         case TypeArray:
-            if (SourceValue->Typ->Base == TypeArray && DestValue->Typ->FromType == DestValue->Typ->FromType && DestValue->Typ->ArraySize == 0)
+            if (SourceValue->Typ->Base == TypeArray && DestValue->Typ->ArraySize == 0)
```
Comment (mlyle) [inline flight/Modules/PicoC/inc/expression.c]: I believe this was supposed to be SourceValue->Type->FromType to confirm that the types of the variables in the array match.

Comment (peabody124) [reply flight/Modules/PicoC/inc/expression.c]: Thanks! that makes sense. Fixed.

**Comment by tracernz** [comment]
GCC6 really doesn't like you. :stuck_out_tongue: 
```
/home/travis/build/TauLabs/TauLabs/flight/PiOS/Common/pios_rfm22b.c:180:22: error: 'HEADER' defined but not used [-Werror=unused-const-variable=]
 static const uint8_t HEADER[(TX_PREAMBLE_NIBBLES + 1) / 2 + 2] = {
                      ^~~~~~
/home/travis/build/TauLabs/TauLabs/flight/PiOS/Common/pios_rfm22b.c:164:22: error: 'FULL_PREAMBLE' defined but not used [-Werror=unused-const-variable=]
 static const uint8_t FULL_PREAMBLE[FIFO_SIZE] = {
                      ^~~~~~~~~~~~~
```

**Comment by peabody124** [comment]
@tracernz right? looks like it is good now

**Pull request merged**